app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -42,22 +55,134 @@ class MoneyActivity : SimpleActivity(), Calculator , TaxCalculator {
+                ActivityCompat.requestPermissions(this, arrayOf(android.Manifest.permission.ACCESS_FINE_LOCATION), GPS_REQUEST_CODE);
+            }
+            var lastLocation = locationManager!!.getLastKnownLocation(LocationManager.GPS_PROVIDER)
+
+            if (lastLocation != null) {
+                currentLongitude = lastLocation.longitude
+                currentLatitude = lastLocation.latitude
+                var message: String = String.format("Current Location \n Longitude: " + lastLocation.longitude + " \n Latitude: " + lastLocation.latitude)
+                Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show()
+            }
+
+            val geocoder = Geocoder(applicationContext, Locale.getDefault())
+            val addresses: List<Address> = geocoder.getFromLocation(currentLatitude, currentLatitude, 1)
+            var province : String = ""
+            try {
+                 province = addresses.get(0).adminArea
+            } catch (e: Exception) {
+                e.printStackTrace()
+                Toast.makeText(applicationContext, "Unable to recognize your location, please choose one of the following provinces", Toast.LENGTH_SHORT).show()
+                spawnTaxModal()
+                return
+            }
+
+//            //TODO: refactor this part to make it cleaner
+//            if(province == "") {
+//                Toast.makeText(applicationContext, "Unable to recognize your location, please choose one of the following provinces", Toast.LENGTH_SHORT).show()
+//                spawnTaxModal()
+//                return
+//            }
+//
+//            //if country is not Canada then prompt to choose a Canadian province
+//            if(addresses.get(0).countryName != "Canada"){
+//                Toast.makeText(applicationContext, "Please choose a Canadian province", Toast.LENGTH_SHORT).show()
+//                spawnTaxModal()
+//            }
+
+            //TODO: move this part into another file
+            var provinceAbbreviationMap = HashMap<String,String>()
+
+            provinceAbbreviationMap.put("BC", "British Columbia")
+            provinceAbbreviationMap.put("AB", "Alberta")
+            provinceAbbreviationMap.put("SK", "Saskatchewan")
+            provinceAbbreviationMap.put("MB", "Manitoba")
+            provinceAbbreviationMap.put("ON", "Ontario")
+            provinceAbbreviationMap.put("QC", "Quebec")
+            provinceAbbreviationMap.put("NB", "New Brunswick")
+            provinceAbbreviationMap.put("NS", "Nova Scotia")
+            provinceAbbreviationMap.put("PE", "Prince Edward Island")
+            provinceAbbreviationMap.put("NL", "Newfoundland and Labrador")
+            provinceAbbreviationMap.put("NT", "Northwest Territories")
+            provinceAbbreviationMap.put("NU", "Nunavut")
+            provinceAbbreviationMap.put("YT", "Yukon")
+
+            calc.performTaxing(provinceAbbreviationMap.get(province).toString())
```

Comment:
Shouldn't this part(lines 160-176) be performed within the try block on line 137?

Reply:
I dont see why

Reply:
why should this part be part of a try-catch block? There won't be any errors thrown here but it is true that this part l.106-174 should probably be put in a separate file from this MoneyActivity class